lib/ui/home/home_view_model.dart

```diff
@@ -1,32 +1,102 @@
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:rxdart/rxdart.dart';
 import 'package:survey_flutter_ic/di/provider/di.dart';
+import 'package:survey_flutter_ic/extension/date_extension.dart';
 import 'package:survey_flutter_ic/model/profile_model.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
 import 'package:survey_flutter_ic/ui/home/home_view_state.dart';
 import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
 import 'package:survey_flutter_ic/usecase/get_profile_use_case.dart';
+import 'package:survey_flutter_ic/usecase/get_survey_use_case.dart';
+
+const _defaultFirstPageIndex = 1;
+const _defaultPageSize = 10;
 
 final homeViewModelProvider =
     StateNotifierProvider.autoDispose<HomeViewModel, HomeViewState>(
         (_) => HomeViewModel(
               getIt.get<GetProfileUseCase>(),
+              getIt.get<GetSurveysUseCase>(),
             ));
 
+final isLoadingProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).isLoading);
+
+final todayProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).today);
+
+final profileAvatarProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).profileAvatar);
+
+final surveysProvider = StreamProvider.autoDispose(
+    (ref) => ref.watch(homeViewModelProvider.notifier).surveys);
+
 class HomeViewModel extends StateNotifier<HomeViewState> {
   final GetProfileUseCase _getProfileUseCase;
+  final GetSurveysUseCase _getSurveysUseCase;
+
+  HomeViewModel(
+    this._getProfileUseCase,
+    this._getSurveysUseCase,
+  ) : super(const HomeViewState.init());
+
+  final BehaviorSubject<bool> _isLoading = BehaviorSubject();
+
+  Stream<bool> get isLoading => _isLoading.stream;
 
-  HomeViewModel(this._getProfileUseCase) : super(const HomeViewState.init());
+  final BehaviorSubject<String> _today = BehaviorSubject();
 
-  Future getProfile() async {
-    state = const HomeViewState.loading();
+  Stream<String> get today => _today.stream;
 
-    final result = await _getProfileUseCase.call();
+  final BehaviorSubject<String> _profileAvatar = BehaviorSubject();
+
+  Stream<String> get profileAvatar => _profileAvatar.stream;
+
+  final BehaviorSubject<List<SurveyModel>> _surveys = BehaviorSubject();
+
+  Stream<List<SurveyModel>> get surveys => _surveys.stream;
+
+  void loadData() async {
+    final profileStream = Stream.fromFuture(_getProfileUseCase.call());
```

Comment:
OPTIONAL: We could use `.asStream()` to convert Future to Stream for usage

```suggestion
    final profileStream = _getProfileUseCase.call().asStream();
```

Reply:
Good point 👍 @hoangnguyen92dn, Thanks for reminding me 🙏